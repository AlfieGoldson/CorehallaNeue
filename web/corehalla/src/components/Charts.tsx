// Library imports
import React, { FC } from 'react';

interface ChartProps {
    width?: string;
    height?: string;
    amount: number;
    bg?: string;
    fg?: string;
}

type Chart = FC<ChartProps>;

export const BarChart: Chart = ({
    width = '100%',
    height = '0.25rem',
    amount = 0,
    bg = 'var(--bg-2)',
    fg = 'var(--accent)',
}: ChartProps) => {
    return (
        <svg width={width} height={height} viewBox="0 0 20 20" preserveAspectRatio="none">
            <rect x="0" y="0" width="20" height="20" fill={bg} />
            <rect x="0" y="0" width={(amount / 100) * 20} height="20" fill={fg} />
        </svg>
    );
};

export const PieChart: Chart = ({
    width = '2rem',
    height = '2rem',
    amount = 0,
    bg = 'var(--bg-2)',
    fg = 'var(--accent)',
}: ChartProps) => {
    return (
        <svg width={width} height={height} viewBox="0 0 20 20">
            <circle r="10" cx="10" cy="10" fill={bg} />
            <circle
                r="5"
                cx="10"
                cy="10"
                fill="transparent"
                stroke={fg}
                strokeWidth="10"
                strokeDasharray={`calc(${amount} * 31.4 / 100) 31.4`}
                transform="rotate(-90) translate(-20)"
            />
        </svg>
    );
};
