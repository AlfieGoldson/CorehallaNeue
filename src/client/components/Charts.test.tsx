import React from 'react';
import { shallow } from 'enzyme';
import { BarChart } from './Charts';

describe('BarChart', () => {
    it('should render correctly', () => {
        const component = shallow(<BarChart amount={10} />);
        expect(component).toMatchSnapshot();
    });
    it('should render background correctly', () => {
        const component = shallow(<BarChart amount={12} bg="#123456" />);
        expect(component).toMatchSnapshot();
    });
    it('should render foreground correctly', () => {
        const component = shallow(<BarChart amount={18} fg="#789456" />);
        expect(component).toMatchSnapshot();
    });
    it('should render width correctly', () => {
        const component = shallow(<BarChart amount={99} width="100px" />);
        expect(component).toMatchSnapshot();
    });
    it('should render height correctly', () => {
        const component = shallow(<BarChart amount={15} height="100px" />);
        expect(component).toMatchSnapshot();
    });
});
