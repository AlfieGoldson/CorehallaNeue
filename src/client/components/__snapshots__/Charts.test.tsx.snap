// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart should render background correctly 1`] = `
<svg
  height="0.25rem"
  preserveAspectRatio="none"
  viewBox="0 0 20 20"
  width="100%"
>
  <rect
    fill="#123456"
    height="20"
    width="20"
    x="0"
    y="0"
  />
  <rect
    fill="var(--accent)"
    height="20"
    width={2.4}
    x="0"
    y="0"
  />
</svg>
`;

exports[`BarChart should render correctly 1`] = `
<svg
  height="0.25rem"
  preserveAspectRatio="none"
  viewBox="0 0 20 20"
  width="100%"
>
  <rect
    fill="var(--bg)"
    height="20"
    width="20"
    x="0"
    y="0"
  />
  <rect
    fill="var(--accent)"
    height="20"
    width={2}
    x="0"
    y="0"
  />
</svg>
`;

exports[`BarChart should render foreground correctly 1`] = `
<svg
  height="0.25rem"
  preserveAspectRatio="none"
  viewBox="0 0 20 20"
  width="100%"
>
  <rect
    fill="var(--bg)"
    height="20"
    width="20"
    x="0"
    y="0"
  />
  <rect
    fill="#789456"
    height="20"
    width={3.5999999999999996}
    x="0"
    y="0"
  />
</svg>
`;

exports[`BarChart should render height correctly 1`] = `
<svg
  height="100px"
  preserveAspectRatio="none"
  viewBox="0 0 20 20"
  width="100%"
>
  <rect
    fill="var(--bg)"
    height="20"
    width="20"
    x="0"
    y="0"
  />
  <rect
    fill="var(--accent)"
    height="20"
    width={3}
    x="0"
    y="0"
  />
</svg>
`;

exports[`BarChart should render width correctly 1`] = `
<svg
  height="0.25rem"
  preserveAspectRatio="none"
  viewBox="0 0 20 20"
  width="100px"
>
  <rect
    fill="var(--bg)"
    height="20"
    width="20"
    x="0"
    y="0"
  />
  <rect
    fill="var(--accent)"
    height="20"
    width={19.8}
    x="0"
    y="0"
  />
</svg>
`;
